Dynatrace/dynatrace-operator
# Adds readonly csi feature-flag + related webhook changes

# Description
Adds the `feature.dynatrace.com/injection-readonly-volume` feature-flag

Which causes the webhook oneagent injection change in a way that the csi-volume can be `readOnly:True`

The following additional volumes and volume-mounts are added in case the feature-flag is enabled:
```
# init container
volumeMounts:
- mountPath: /mnt/agent-conf
  name: oneagent-agent-conf
...
# app container
volumeMounts:
  - mountPath: /opt/dynatrace/oneagent-paas/agent/conf
    name: oneagent-agent-conf
...
volumes:
- emptyDir: {}
  name: oneagent-agent-conf
```

```
# app container
volumeMounts:
  - mountPath: /opt/dynatrace/oneagent-paas/datastorage
    name: oneagent-datastorage
...
volumes:
- emptyDir: {}
  name: oneagent-datastorage
```

```
# app container
volumeMounts:
  - mountPath: /opt/dynatrace/oneagent-paas/log
    name: oneagent-log
...
volumes:
- emptyDir: {}
  name: oneagent-log
```

+1 additional envvar is also added to the injected init-container: (this is always added, just not set to true by default)
```
- name: CSI_VOLUME_READONLY
   value: "true"
```

## How can this be tested?
When the feature-flag is set, the injected pod should be able to start, but the agent will not work properly, that will be added in a follow-up.
Checking the deployed Pod's injected yaml, to verify if mentioned changes are present.


## Checklist
